core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooReader.java

```diff
@@ -121,6 +123,36 @@ public boolean exists(String zPath, Watcher watcher)
     return getStatus(zPath, watcher) != null;
   }
 
+  /**
+   * read the ZooKeeper acls for a node.
+   */
+  public List<ACL> getAcls(String zPath, Stat stat) throws KeeperException, InterruptedException {
+    requireNonNull(stat); // required for ZooKeeper 3.4
+    return retryLoop(zk -> zk.getACL(zPath, stat));
+  }
+
+  public void recursiveAclRead(String parentPath, Map<String,List<ACL>> acls) {
+    try {
+
+      Stat dummy = new Stat(); // required form ZooKeeper 3.4 compatibility, unused here.
+      acls.put(parentPath, getAcls(parentPath, dummy));
+
+      for (String child : getChildren(parentPath)) {
+        String childPath = parentPath + "/" + child;
+        List<ACL> acl = getAcls(childPath, dummy);
+        acls.put(childPath, acl);
+        recursiveAclRead(childPath, acls);
```

Comment:
You could alternatively use `ZKUtil.visitSubTreeDFS`.  See Upgrader9To10.validateACLs for an example.